app/subscriber.py

```diff
@@ -127,6 +120,99 @@ def offline_receipt_to_case(message: Message):
     log.info('Message processing complete')
 
 
+def ppo_undelivered_mail_to_case(message: Message):
+    log = logger.bind(message_id=message.message_id,
+                      subscription_name=PPO_UNDELIVERED_SUBSCRIPTION_NAME,
+                      subscription_project=PPO_UNDELIVERED_SUBSCRIPTION_PROJECT_ID)
+
+    log.debug('Pub/Sub Message received for processing')
+
+    payload = validate_message(message.data, log, ['caseRef', 'productCode'])
+    if not payload:
+        return  # Failed validation
+
+    case_ref, product_code, date_time = payload['caseRef'], payload['productCode'], payload['dateTime']
+
+    log = log.bind(case_ref=case_ref, created=date_time, product_code=product_code)
+
+    receipt_message = {
+        'event': {
+            'type': 'UNDELIVERED_MAIL_REPORTED',
+            'source': 'RECEIPT_SERVICE',
+            'channel': 'PPO',
+            'dateTime': date_time,
+            'transactionId': str(uuid.uuid4())
```

Comment:
we shouldn't be creating a new transactionId here, there should be one coming in on the message from GCP pub/sub message

Reply:
there is not

Reply:
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?pageId=20189919